scripts/cherry-pick-patch-commits.js

```diff
@@ -0,0 +1,164 @@
+ * cherry-picks that result in conflicts.
+ *
+ * Note that this does not create a branch - you will be in detached HEAD state. You can create a branch afterwards
+ * if you want, via `git checkout -b <branchname>`.
+ */
+
+const args = process.argv.slice(2);
+
+const parser = require('conventional-commits-parser');
+const parserOpts = require('conventional-changelog-angular/parser-opts');
+const simpleGit = require('simple-git/promise')();
+const {spawnSync} = require('child_process');
+
+const CONFLICT_MESSAGE = 'after resolving the conflicts, mark the corrected paths';
+
+// Resolves to the most recent non-pre-release git tag.
+async function getMostRecentTag() {
+  const tags = await simpleGit.tags();
+  // Filter old independent-versioned tags and pre-releases
+  const filteredTags = tags.all.filter((tag) => /^v/.test(tag) && !/-/.test(tag));
+  return filteredTags[filteredTags.length - 1];
+}
+
+// Resolves to an array of commits after the given tag, from earliest to latest (for proper cherry-picking).
+async function getCommitsAfterTag(tag) {
+  if (!args.includes('--no-fetch')) {
+    await simpleGit.fetch();
+  }
+  const log = await simpleGit.log({
+    from: tag,
+    to: 'origin/master',
+    // simple-git defaults to only including subject + tag/branch (%s%d) in message, which adds noise and is not useful
+    // for finding BREAKING CHANGEs (which are in body, %b).
+    format: {
+      hash: '%H', // Contains full hash only, for cherry-picking
+      message: '%s\n\n%b', // Contains subject + body for passing to conventional-commits-parser
+      oneline: '%h %s' // Contains abbreviated hash + subject for summary lists
+    },
+    // Use a splitter that is very unlikely to be included in commit descriptions
+    splitter: ';;;'
+  });
+  return log.all.reverse();
+}
+
+// Returns true if commit should NOT be cherry-picked.
+function shouldSkipCommit(logLine) {
+  const parsedCommit = parser.sync(logLine.message, parserOpts);
+  return parsedCommit.type === 'feat' || // feature commit
+    parsedCommit.notes.find((note) => note.title === 'BREAKING CHANGE') || // breaking change commit
+    (parsedCommit.type === 'chore' && parsedCommit.subject === 'Publish'); // Publish (version-rev) commit
+}
+
+// Checks out the given tag and attempts to cherry-pick each commit in the list against it.
+// If a conflict is encountered, that cherry-pick is aborted and processing moves on to the next commit.
+async function attemptCherryPicks(tag, list) {
+  const results = {
+    successful: [],
+    conflicted: [],
+    skipped: [],
+  }
+
+  console.log(`Checking out ${tag}`)
+  await simpleGit.checkout([tag]);
+
+  for (let logLine of list) {
+    if (shouldSkipCommit(logLine)) {
+      results.skipped.push(logLine);
+      continue;
+    }
+
+    try {
+      await simpleGit.raw(['cherry-pick', '-x', logLine.hash]);
+      results.successful.push(logLine);
+    } catch (e) {
+      if (e.message.includes(CONFLICT_MESSAGE)) {
```

Comment:
This part confuses me. Where does this error message come from? Is it from the git cherry-pick command?

Reply:
Yes, it's from the command.

> If the process times out or has a non-zero exit code, this method will throw an Error that will include the full result of the underlying child_process.spawnSync().